From bbd48a5613c872883616884cfaf41665b0e4ec9b Mon Sep 17 00:00:00 2001
From: Ralph Giles <giles@mozilla.com>
Date: Fri, 10 Feb 2017 12:58:18 -0800
Subject: [PATCH] Bug 1338655 - Don't try to build mp4parse bindings. r=froydnj

We use the cheddar crate to generate a C header file
for our mp4parse_capi wrapper crate. Currently we
do this at code check-in time via update-rust.sh.

Cargo 0.18 and later will try to execute a build.rs
file in the crate source tree regardless of whether
it's specified in Cargo.toml so patching out that
line just results in 'crate cheddar not found'.

This change restores the old behaviour by substituting
a 'build = false' line instead.

We do have syntex vendored, but we don't currently build
it by default, so I prefer this solution to just vendoring
cheddar and generating the header at build time. The syntex
crate is quite large and adds significantly to our compile
time.

MozReview-Commit-ID: InJRRODWAdP

--HG--
extra : rebase_source : 29378fcbc86015ce6cc22dc66d38a43ddbac204e
---
 media/libstagefright/binding/mp4parse-cargo.patch     | 5 +++--
 media/libstagefright/binding/mp4parse_capi/Cargo.toml | 2 ++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/binding/mp4parse_capi/Cargo.toml b/media/libstagefright/binding/mp4parse_capi/Cargo.toml
index aee7ee947151a27c..d7e3f55119d3f4b6 100644
--- a/media/libstagefright/binding/mp4parse_capi/Cargo.toml
+++ b/media/libstagefright/binding/mp4parse_capi/Cargo.toml
@@ -18,6 +18,8 @@ exclude = [
   "*.mp4",
 ]
 
+build = false
+
 [dependencies]
 byteorder = "1.0.0"
 "mp4parse" = {version = "0.6.0", path = "../mp4parse"}
-- 
2.12.2

